src/wx/config/strutils-test.cpp

```diff
@@ -118,10 +118,25 @@ TEST(UTF16ToUTF8Test, MultiByte) {
     EXPECT_EQ(vec[2], 0xAC);
 }
 
+TEST(UTF16ToUTF8Test, DualMultiByte) {
+    // This is a variant of the above to test the buffer reset is done properly.
+    std::vector<char16_t> utf16 = {0x20AC, 0x20AC, 0};
+    auto vec = config::utf16_to_utf8_vector(utf16.data());
+
+    ASSERT_EQ(vec.size(), 6);
+
+    EXPECT_EQ(vec[0], 0xE2);
+    EXPECT_EQ(vec[1], 0x82);
+    EXPECT_EQ(vec[2], 0xAC);
+    EXPECT_EQ(vec[3], 0xE2);
+    EXPECT_EQ(vec[4], 0x82);
+    EXPECT_EQ(vec[5], 0xAC);
```

Comment:
Assuming you made the return type `std::string` (see below), you can just do this here:

```cpp
auto str = config::utf16_to_utf8_str(utf16);

ASSERT_EQ(str.size(), 6);

EXPECT_EQ(str, "LITERAL-UTF8-HERE");
```
. No need to even add a BOM to the source file, this will work fine with everything.